src/main/java/org/terasology/taskSystem/TaskManagementSystem.java

```diff
@@ -505,10 +505,10 @@ private Region3i getBuildingExtents(BuildingType buildingType, Region3i region)
         int maxZ = Integer.MIN_VALUE;
 
         for (SpawnBlockRegionsComponent.RegionToFill regionToFill : regionsToFill) {
-            minX = Math.min(minX, regionToFill.region.getMinX());
-            maxX = Math.max(maxX, regionToFill.region.getMaxX());
-            minZ = Math.min(minZ, regionToFill.region.getMinZ());
-            maxZ = Math.max(maxZ, regionToFill.region.getMaxZ());
+            minX = Math.min(minX, regionToFill.region.minX());
+            maxX = Math.max(maxX, regionToFill.region.maxX());
+            minZ = Math.min(minZ, regionToFill.region.minZ());
+            maxZ = Math.max(maxZ, regionToFill.region.maxZ());
         }
```

Comment:
Just as a side note here: This builds the union over all `regionsToFill`. This could be achieved by

```java
BlockRegion buildingExtents = regionsToFill.stream()
  .map(r->r.region)
  .reduce(new BlockRegion(BlockRegion.INVALID), BlockRegion::union, BlockRegion::union);
```

I admit that this is probably hard to read, but maybe this would be okay?

```java
BlockRegion buildingExtents = BlockRegion.encompassRegions(regionsToFill.stream().map(r->r.region));
```